src/Catalog/Helpers/NonhijackableV2HttpMessageHandlerWrapper.cs

```diff
@@ -0,0 +1,103 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net.Http;
+using System.Reflection;
+using System.Text.RegularExpressions;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace NuGet.Services.Metadata.Catalog.Helpers
+{
+    /// <summary>
+    /// A <see cref="HttpMessageHandler"/> that wraps another <see cref="HttpMessageHandler"/> and prevents queries to the V2 feed from being hijacked by the search service.
+    /// </summary>
+    public class NonhijackableV2HttpMessageHandlerWrapper : HttpMessageHandler
```

Comment:
>NonhijackableV2HttpMessageHandlerWrapper [](start = 17, length = 40)

NonHijackedV2HttpMessageHandler

the "able" and "wrapper" parts are superfluous.

Inherit DelegatingHandler since it is meant for this kind of thing.